### PR TITLE
Use new avatar API service for avatar uploads

### PR DESCRIPTION
**Pull request by nicl**
Use the new API. Change is behind a switch.

This is **not** ready to merge yet (the new Avatar API is not yet prod-ready - although as it is behind a switch there isn't much danger), but as I am a complete noob when it comes to client-side stuff I wanted to get early feedback on this.

See commit messages for further details.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/account-profile.js [hunk 1]
```diff
@@ -34,11 +39,14 @@ define([
...
-            noServerError: 'The image upload server could not be reached.'
+            noServerError: 'Oops, the Avatar upload service is currently unavailable. Please try again shortly.',
+            avatarUploadSuccess: 'Thank you for uploading your avatar. It will be checked by Guardian moderators shortly.',
+            avatarUploadFailure: 'Oops, something went wrong. Please try again.'
         };
 
         self.urls = {
-            avatarTokenUrl: 'https://gu-image-upload.appspot.com/upload-endpoint-generator'
+            avatarTokenUrl: 'https://gu-image-upload.appspot.com/upload-endpoint-generator',
+            avatarApiUrl: 'https://avatar.code.dev-guardianapis.com/v1/avatars'
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/identity/account-profile.js]: I'm nervous about hard coding endpoints, we should really have it in frontend.properties and read from there

**Comment by johnduffell** [comment]
looks pretty good although I'm not an expert - maybe the proof of the pudding is in the eating/testing.  You might want to fix up your build at some point:
```
Multiple line break at static/src/javascripts/projects/common/modules/identity/account-profile.js :
[11:29:11]   154 |        xhr.send(formData);
[11:29:11]   155 |    };
[11:29:11]   156 |
[11:29:11]--------^
[11:29:11]   157 |
[11:29:11]   158 |    /*
[11:29:11]>> 1 code style errors found!
```
:+1:

File: static/src/javascripts/projects/common/modules/identity/account-profile.js (hunk 1 above)
Comment (nicl) [inline static/src/javascripts/projects/common/modules/identity/account-profile.js]: Thanks - I did wonder what the right way to do this was. Will update.

**Comment by johnduffell** [comment]
if we're not shipping this soon we should probably close it for now

**Comment by nicl** [comment]
Okay, going to close as we aren't ready yet with PROD Avatar API and I'm reluctant to merge in until then.

**Pull request closed without merging**